### ISSUE TITLE
Reward points are not subtracted when order is placed.

### ISSUE THREAD

**Issue by vyshemirsky** [opencart/opencart#3637]
Calculations in totals are correct, reward points are applied and the price discounted according to the amount of claimed products. However, once the order is placed the reward points are not subtracted from customer account, and can be used again and again and again.

version: 2.1.0.1

**Comment by vyshemirsky** [comment]
And this happens when the products provide new reward points at the same time as being partially paid with previous reward points.

**Comment by danielkerr** [comment]
do some reasearch

**Comment by vyshemirsky** [comment]
Thank you, very informative answer.

**Comment by stockmaster** [comment]
@vyshemirsky Is the order set to "completed" at the stage you looking? AS most of the opencart engine, some calculations will only be triggered on certain order "status". Do a couple tests in that regard

**Comment by vyshemirsky** [comment]
Yes. The reward points are not subtracted when the order is set to "completed" either.

**Comment by stockmaster** [comment]
i personally dont use reward points so was just a thought. Will try in the sandbox

**Comment by vyshemirsky** [comment]
Adding new reward points works from orders management page, but even once the order is switched to completed nothing happens.

The payment system I use is "Bank transfer", the totals work with rewards just fine. Orders are placed correctly. I can add new rewards from the orders page.

My normal lifecycle is pending->processing->processed->shipped->completed. 

No points subtracted after any of the above stages.

**Comment by vyshemirsky** [comment]
I am not alone who has the problem, here are two more users:
http://forum.opencart.com/viewtopic.php?f=191&t=137332&p=547009&hilit=reward#p547009

**Comment by stockmaster** [comment]
Ok seems a good due diligent step by step explanation. maybe Daniel can look into it this way.

**Comment by stockmaster** [comment]
well notice though, there is said "the points are NOT deducted from the customers total points until the admin changes the order status to complete" this is exactly what i was referring to. 
but that a different problem, the problem being that 1 customer can make 2 simultaneous orders, pay them, using the same reward points.

**Comment by vyshemirsky** [comment]
Fair enough. I do agree with those two who say it is inconvenient, but that is not relevant to the current issue, so I deleted my reference to that forum thread.

My issue is that points are never subtracted at all.

**Comment by stockmaster** [comment]
Right. as i said i do not 100% familiar with reward points. But they Should be processed (deducted) when order is complete. I am unsure however if you can change their deduction to a "processed" status instead. Only daniel can answer this, or any other user very familiar with reward points processes.

**Comment by vyshemirsky** [comment]
Tried going directly Pending -> Complete
Still no effect.

**Comment by stockmaster** [comment]
will try to look into it, but if Daniel can look i guess it would be more effective. Meanwhile any chance you can try a sandbox install with latest RC from github ? as you using probably Release not updated?

**Comment by vyshemirsky** [comment]
I am using a fresh install from 2.1.0.1 release (not updated)

**Comment by stockmaster** [comment]
this would only help to know if that "bug", if there's any, was already addressed in some commit

**Comment by vyshemirsky** [comment]
More information for Daniel to look at.
Clean install using a new MySQL database on a new server. Version 2.1.0.1 without anything else, directly from the downloads server. No modifications to the example setup.

Registered as a new customer.
Created first order, bought 'MacBook'

As admin, changed the order from 'Pending' to 'Complete'. Then clicked on a button to add reward points.

As a registered customer created the second order.
Bought "Apple Cinema 30"" quantity 2
Went to shopping cart and claimed all 400 points earned with the first order.

As admin, changed the order from 'Pending' to 'Complete'. No change in customers reward point balance.
Clicked on adding 200 points from the second order. Customer's balance became 600.

**Comment by vyshemirsky** [comment]
master branch does not have this error, but a lot of other things fail - master branch does not install, so I switched to a version from 14-Nov-2015, and this version throws a lot of warnings in admin interface, and does not show 'my account' in customer interface, so it is not an option for me to update the shop to the master branch.

**Comment by vyshemirsky** [comment]
OK, resolved...

diff -r opencart-2.1.0.1/upload/catalog/model/total/reward.php opencart-master/upload/catalog/model/total/reward.php
68c68
<               if ($this->model_account_customer->getRewardTotal($order_info['customer_id']) < $points) {
---
>               if ($this->model_account_customer->getRewardTotal($order_info['customer_id']) >= $points) {

**Comment by stockmaster** [comment]
Could you rephrase please, or make more clear what changes you did to fix the problem? 
Thank you (code is a little dirty in your post)

**Comment by sharanhayer** [comment]
@stockmaster the comparison string was incorrect ( better say was opposite as required) now its >= before it was <= , thats ehy it was detecting all orders with reward points as fraud

**Comment by stockmaster** [comment]
well, i didnt have a chance to test however from what @vyshemirsky  says, that change made that the points are then NOT deducted from customer reward points on Complete.
Ence why i requested Daniel to reverify

**Comment by vyshemirsky** [comment]
Yes, the points weren't deducted at any stage because the order was marked as fraud. The change is correct in the master branch.

**Comment by stockmaster** [comment]
Gotcha! so you updated your install by @sharanhayer commit! sorry i understood it the other way arround! Great TX!

**Comment by stockmaster** [comment]
Will Delete above COmment to not confuse things then! Thank you both

**Comment by harron444** [comment]
@vyshemirsky  How can I get the source code for this file opencart-master/upload/catalog/model/total/reward.php
